### DISCUSSION TITLE
restart with distributed mesh?

### DISCUSSION THREAD

**Discussion by heinono1** [idaholab/moose#22632] [Q&A General] [answered]
I run rather large simulations and I need to do an initial run in which the system equilibrates. After that, I need to read in the equilibrated solution to seed a new calculation that has added driving forces. I know how to do this using replicated meshes. However, I need to use distributed (parallel) meshes in order to avoid running out of memory. The descriptions of restart (Variable Initialization or using SolutionUserObject) indicate that these methods only work for replicated meshes. Is this still true or is there a capability do do these restarts with distributed mesh?

Thanks.

**Comment by GiudGiud** [comment, marked as answer]
For distributed meshes you should look at checkpoint restart.

I dont know about solutionUO. @friedmud or @permcody ?

**Reply by heinono1** [reply]
Hi Guillaume. Yes, I routinely use checkpoint but my understanding is that with checkpointing you are limited to using the exact same variables as in the prior runs. Is that not the case?

**Reply by GiudGiud** [reply]
It is the case. 
@friedmud is working on a more flexible checkpoint

**Reply by heinono1** [reply]
OK - thanks!
